testutil/network/gatewaynet/network.go

```diff
@@ -0,0 +1,190 @@
+	require.NotNil(t, resp)
+	require.NotNil(t, resp.TxHash)
+	require.Equal(t, uint32(0), resp.Code)
+}
+
+// Start initializes the in-memory network and performs the following setup:
+//   - populates a new in-memory keyring with a sufficient number of pre-generated accounts.
+//   - configures the application module's genesis state using addresses corresponding
+//     to config.NumApplications number of the same pre-generated accounts which were
+//     added to the keyring.
+//   - configures the supplier module's genesis state using addresses corresponding to
+//     config.NumSuppliers number of the same pre-generated accounts which were added
+//     to the keyring.
+//   - creates the on-chain accounts in the accounts module which correspond to the
+//     pre-generated accounts which were added to the keyring.
+func (memnet *inMemoryNetworkWithGateways) Start(_ context.Context, t *testing.T) {
+	t.Helper()
+
+	// Application module genesis state fixture data generation is independent
+	// of that of the supplier module.
+	if memnet.Config.AppSupplierPairingRatio > 0 {
+		panic("AppSupplierPairingRatio must be 0 for inMemoryNetworkWithGateways, use NumApplications instead")
+	}
+
+	memnet.InitializeDefaults(t)
+	memnet.CreateKeyringAccounts(t)
+
+	// Configure gateway and application module genesis states.
+	memnet.configureGatewayModuleGenesisState(t)
+	memnet.configureAppModuleGenesisState(t)
+
+	memnet.Network = network.New(t, *memnet.GetCosmosNetworkConfig(t))
+
+	memnet.CreateOnChainAccounts(t)
+}
+
+// TODO_IN_THIS_COMMIT: fix comment...
+// GatewayModuleGenesisStateWithAddresses generates a GenesisState object with
+// a gateway list full of gateways with the given addresses.
+func (memnet *inMemoryNetworkWithGateways) configureGatewayModuleGenesisState(t *testing.T) {
+	t.Helper()
+
+	gatewayGenesisState := gatewaytypes.DefaultGenesis()
+	for gatewayIdx := 0; gatewayIdx < memnet.Config.NumGateways; gatewayIdx++ {
+		stake := sdk.NewCoin("upokt", sdk.NewInt(int64(gatewayIdx)))
+		preGeneratedAcct, ok := memnet.PreGeneratedAccountIterator.Next()
+		require.Truef(t, ok, "pre-generated accounts iterator exhausted")
+		require.Truef(t, ok, "pre-generated accounts iterator exhausted")
+
+		gateway := gatewaytypes.Gateway{
+			Address: preGeneratedAcct.Address.String(),
+			Stake:   &stake,
+		}
+
+		// TODO_CONSIDERATION: Evaluate whether we need `nullify.Fill` or if we should enforce `(gogoproto.nullable) = false` everywhere
+		// nullify.Fill(&gateway)
+		gatewayGenesisState.GatewayList = append(gatewayGenesisState.GatewayList, gateway)
+	}
+
+	gatewayGenesisBuffer, err := memnet.GetCosmosNetworkConfig(t).Codec.MarshalJSON(gatewayGenesisState)
+	require.NoError(t, err)
+
+	// Add supplier module genesis supplierGenesisState to the network config.
+	memnet.GetCosmosNetworkConfig(t).GenesisState[gatewaytypes.ModuleName] = gatewayGenesisBuffer
+}
+
```

Comment:
```suggestion

// configureAppModuleGenesisState takes from the memnet configuration
// the number of applicationns and creates gateway actors and "stakes"
// them in the ApplicationGenesisState before serialiazing this and
// adding the serialized ApplicationGenesisState to the memnet network
// config.
```

Reply:
Can I ask why no godoc-like comment explaining this function (even if its not exported)?